src/xcode/ENA/ENA/Source/Services/Risk/Provider/Helper/CachedAppConfiguration.swift

```diff
@@ -157,72 +157,74 @@ final class CachedAppConfiguration: AppConfigurationProviding {
 
 	private func getAppConfig(with etag: String? = nil) -> Future<AppConfigResponse, Never> {
 		return Future { promise in
-			Self.queue.sync(flags: .barrier) {
+			Self.queue.sync {
```

Comment:
Will this 'Return immediately...' case still happen if a serial queue is used now?
Is the property `requestIsRunning` still used, I can't see any change?

Reply:
`requestIsRunning` is still used and will return early and save the promise for later fulfillment.
I think `requestIsRunning` is still needed with the serial queue and even if its called in sync. Because the call to `fetchAppConfiguration` is escaping. The serial queue can be empty while `fetchAppConfiguration` is still executing.